app/services/users/register_user_service.rb

```
@@ -0,0 +1,163 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2020 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Users
+  class RegisterUserService
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    def call
+      %i[
+        register_invited_user
+        register_ldap_user
+        ensure_registration_allowed!
+        ensure_user_limit_not_reached!
```

Shouldn't this (i.e. whether the user limit has been reached) be checked before registering a new invited or ldap user? Depending on the user limit fail fast settings there can be more invitations than could be activated due to the user limit IIRC.

---

Yes I agree, not sure why this was not added first